### ISSUE TITLE
Inconsistent behavior of pathlib.WindowsPath with drive paths

### ISSUE THREAD

**Issue by 3aeaa39e-ab54-4e29-a5ec-decbf2a0f0ce** [python/cpython#80486]
BPO | [36305](https://bugs.python.org/issue36305)
--- | :---
Nosy | @pfmoore, @tjguk, @zware, @serhiy-storchaka, @eryksun, @zooba, @kmaork, @nmat
PRs | <li>python/cpython#12361</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-03-15.15:37:05.961>
labels = ['type-bug', '3.8', '3.9', '3.10', 'library', 'OS-windows']
title = 'Inconsistent behavior of pathlib.WindowsPath with drive paths'
updated_at = <Date 2021-03-20.06:55:56.836>
user = 'https://github.com/kmaork'
```

bugs.python.org fields:
```python
activity = <Date 2021-03-20.06:55:56.836>
actor = 'eryksun'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)', 'Windows']
creation = <Date 2019-03-15.15:37:05.961>
creator = 'kmaork'
dependencies = []
files = []
hgrepos = []
issue_num = 36305
keywords = ['patch']
message_count = 18.0
messages = ['337999', '338007', '338011', '338663', '338667', '338669', '338675', '338677', '338680', '338685', '338708', '338727', '338768', '340388', '340396', '340826', '343621', '343784']
nosy_count = 8.0
nosy_names = ['paul.moore', 'tim.golden', 'zach.ware', 'serhiy.storchaka', 'eryksun', 'steve.dower', 'kmaork', 'nmat']
pr_nums = ['12361']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue36305'
versions = ['Python 3.8', 'Python 3.9', 'Python 3.10']
```

</p></details>

### Linked PRs
* gh-102454

**Comment by barneygale** [comment]
This can be solved quite simply once #101667 lands.

**Comment by barneygale** [comment]
I've opened #102454, which brings @kmaork's patch up-to-date and solves the issue (I think) with a small change to pathlib.

@pfmoore @eryksun I appreciate it's been 4 years, but would you be interested in reviewing that PR? Thanking you!

**Comment by barneygale** [comment]
It looks like `ntpath.realpath()` is affected by a similar bug. I've logged a new issue:

- #102475

**Comment by barneygale** [comment]
Resolved in #102454 / [90f1d77](https://github.com/python/cpython/commit/90f1d777177e28b6c7b8d9ba751550e373d61b0a), merged into the 3.12 but not backported.

Thanks again @kmaork for reporting this and working on the initial patch!
